mailbox/event/json/src/main/scala/org/apache/james/event/json/DTOs.scala

```diff
@@ -74,75 +76,72 @@ object DTOs {
     def fromJava(javaMessageMetaData: JavaMessageMetaData): MessageMetaData = DTOs.MessageMetaData(
       javaMessageMetaData.getUid,
       javaMessageMetaData.getModSeq,
-      javaMessageMetaData.getFlags,
+      Flags.fromJavaFlags(javaMessageMetaData.getFlags),
       javaMessageMetaData.getSize,
       javaMessageMetaData.getInternalDate.toInstant,
       javaMessageMetaData.getMessageId)
   }
 
-  case class MessageMetaData(uid: MessageUid, modSeq: Long, flags: JavaMailFlags, size: Long, internalDate: Instant, messageId: MessageId) {
-    def toJava: JavaMessageMetaData = new JavaMessageMetaData(uid, modSeq, flags, size, Date.from(internalDate), messageId)
+  case class MessageMetaData(uid: MessageUid, modSeq: Long, flags: Flags, size: Long, internalDate: Instant, messageId: MessageId) {
+    def toJava: JavaMessageMetaData = new JavaMessageMetaData(uid, modSeq, Flags.toJavaFlags(flags), size, Date.from(internalDate), messageId)
   }
 
-  object Flags {
-    val ANSWERED = "\\Answered"
-    val DELETED = "\\Deleted"
-    val DRAFT = "\\Draft"
-    val FLAGGED = "\\Flagged"
-    val RECENT = "\\Recent"
-    val SEEN = "\\Seen"
-    val ALL_SYSTEM_FLAGS = List(ANSWERED, DELETED, DRAFT, FLAGGED, RECENT, SEEN)
-
-    def toJavaFlags(serializedFlags: Array[String]): JavaMailFlags = {
-      serializedFlags
-        .map(toJavaMailFlag)
-        .foldLeft(new FlagsBuilder)((builder, flag) => builder.add(flag))
-        .build()
-    }
+  case class UserFlag(value: String) extends AnyVal
 
-    def toJavaMailFlag(flag: String): JavaMailFlags = ALL_SYSTEM_FLAGS.contains(flag) match {
-      case true => new FlagsBuilder().add(stringToSystemFlag(flag)).build()
-      case false => new FlagsBuilder().add(flag).build()
-    }
+  object SystemFlag extends Enumeration {
+    type SystemFlag = Value
+    val Answered, Deleted, Draft, Flagged, Recent, Seen = Value
+  }
 
-    def fromJavaFlags(flags: JavaMailFlags): Array[String] = {
-      flags.getUserFlags ++ flags.getSystemFlags.map(flag => systemFlagToString(flag))
+  case class Flags(systemFlags: Seq[SystemFlag], userFlags: Seq[UserFlag])
+
+  object Flags {
+
+    def toJavaFlags(scalaFlags: Flags): JavaMailFlags = {
+      (scalaFlags.systemFlags ++ scalaFlags.userFlags)
+        .foldLeft(new FlagsBuilder)((builder, flag) =>
```

Comment:
Extract the function please

Reply:
Well, extracting means have a type Any as parameter because in Scala you don't have `A|B` construction in the language.
I rewrote it with a foreach.